syntax = "proto3";

service Service{
    rpc sayPublic (sayRequest) returns (sayResponse) {}
    rpc sayUser (sayRequest) returns (sayResponse) {}
    rpc sayAdmin (sayRequest) returns (sayResponse) {}
    rpc sayUnaryStream (sayRequest) returns (stream sayResponse) {}
    rpc sayStreamUnary (stream sayRequest) returns (sayResponse) {}
    rpc sayStreamStream (stream sayRequest) returns (stream sayResponse) {}
}

message sayRequest {
  string name = 1;
}

message sayResponse {
  string reply = 1;
}
